xarray/core/dataset.py

```diff
@@ -5401,6 +5401,13 @@ def transpose(
         numpy.transpose
         DataArray.transpose
         """
+        # Raise error if list is passed as dims
+        if (len(dims) > 0) and (isinstance(dims[0], list)):
+            list_fix = [f"'{x}', " if isinstance(x, str) else f"{x}, " for x in dims[0]]
```

Comment:
If you make it str(x) it prints as "x, y" I wanted it to be "'x', 'y'" to show that the input requires distinct strings. For instance if you pass ['x', 'y', ...] it will show "'x', 'y', Ellipses" without the extra quotes its would be hard to differentiate the string inputs from the ellipses inputs. 

I'll go ahead fix the test thats easy.

Reply:
But I think that python adds the quote — i.e. the currently proposed code instead prints `Expected "\'y\', \'x\'". Received "[\'y\', \'x\']" instead'.`, with the escapes...

Reply:
You're right, of course — I'm confusing `repr` and `str`:

```python
In [4]: print(f"{str('aoue')}")
aoue

In [5]: print(f"{repr('aoue')}")
'aoue'
```

Reply:
Aahh I have never even heard of repr before, thanks. I added that as its much more readable.